PurchasesTests/Purchasing/ProductInfoTests.swift

```diff
@@ -0,0 +1,273 @@
+                                        subscriptionGroup: nil,
+                                        discounts: nil)
+        expect(productInfo.asDictionary()["normal_duration"] as? String) == normalDuration
+    }
+
+    func testAsDictionaryConvertsIntroDurationCorrectlyForFreeTrial() {
+        let trialDuration = "P3M"
+        let productInfo = RCProductInfo(productIdentifier: "cool_product",
+                                        paymentMode: .none,
+                                        currencyCode: "UYU",
+                                        price: 9.99,
+                                        normalDuration: nil,
+                                        introDuration: trialDuration,
+                                        introDurationType: .freeTrial,
+                                        introPrice: nil,
+                                        subscriptionGroup: nil,
+                                        discounts: nil)
+        expect(productInfo.asDictionary()["trial_duration"] as? String) == trialDuration
+        expect(productInfo.asDictionary()["intro_duration"]).to(beNil())
+    }
+
+    func testAsDictionaryConvertsIntroDurationCorrectlyForIntroPrice() {
+        let introDuration = "P3M"
+        let productInfo = RCProductInfo(productIdentifier: "cool_product",
+                                        paymentMode: .none,
+                                        currencyCode: "UYU",
+                                        price: 9.99,
+                                        normalDuration: nil,
+                                        introDuration: introDuration,
+                                        introDurationType: .introPrice,
+                                        introPrice: nil,
+                                        subscriptionGroup: nil,
+                                        discounts: nil)
+        expect(productInfo.asDictionary()["intro_duration"] as? String) == introDuration
+        expect(productInfo.asDictionary()["trial_duration"]).to(beNil())
+    }
+
+    func testAsDictionaryDoesntAddIntroDurationIfDurationTypeNone() {
+        let introDuration = "P3M"
+        let productInfo = RCProductInfo(productIdentifier: "cool_product",
+                                        paymentMode: .none,
+                                        currencyCode: "UYU",
+                                        price: 9.99,
+                                        normalDuration: nil,
+                                        introDuration: introDuration,
+                                        introDurationType: .none,
+                                        introPrice: nil,
+                                        subscriptionGroup: nil,
+                                        discounts: nil)
+        expect(productInfo.asDictionary()["trial_duration"]).to(beNil())
+        expect(productInfo.asDictionary()["trial_duration"]).to(beNil())
```

Comment:
```suggestion
        expect(productInfo.asDictionary()["intro_duration"]).to(beNil())
```

Reply:
I was clearly tired when I got to this file 😂